test/api/api_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+ *    Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package api
+
+import (
+	"context"
+
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/Peripli/service-manager/bootstrap"
+	"github.com/Peripli/service-manager/env"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gavv/httpexpect"
+	. "github.com/onsi/ginkgo"
+)
+
+func TestAPI(t *testing.T) {
+	os.Chdir("../..")
+	RunSpecs(t, "API Tests Suite")
+}
+
+func getServerRouter() *mux.Router {
+	serverEnv := env.New(&env.ConfigFile{
+		Path:   "./test/api",
+		Name:   "application",
+		Format: "yml",
+	}, "SM")
+	srv, err := bootstrap.CreateServer(context.Background(), serverEnv)
+	if err != nil {
+		logrus.Fatal("Error creating server: ", err)
+	}
+	return srv.Router
+}
+
+var _ = Describe("Service Manager API", func() {
+	var testServer *httptest.Server
+
+	BeforeSuite(func() {
+		testServer = httptest.NewServer(getServerRouter())
+		SM = httpexpect.New(GinkgoT(), testServer.URL)
```

Comment:
this is hard to follow, maybe name this file api_suite_test, move the util things here and let testBrokers and testPlatforms be in the broker/platforms_test

Reply:
util things were here initially but this caused cyclic deps